File: leetcode/LongestPalindrome/algo.cpp
class Solution {
public:
    string longestPalindrome(string s) {
        if (s.empty() || s.size() < 1) return "";
        string res;
        int max_len = 0;
        for (int i = 0; i < s.size(); i++) {
            int l = i, r = i;
            //pgryciuk - slide same character
            while(r + 1 < s.size() && s[r+1] == s[i]) {
                r++;
                i++;
            }
            while(l - 1 >= 0 && r + 1 < s.size() && s[l-1] == s[r+1]) {
                r++;
                l--;
            }
            if (r - l + 1 > max_len) {
                max_len = r - l + 1;
                res = s.substr(l, max_len);
            }
        }
        return res;
    }
   
};

File: leetcode/bitwisecomplement/algo.py
class Solution:
    def bitwiseComplement(self, N: int) -> int:
        if N == 0:
            return 1
        
        if N == 1:
            return 0
        
        up = 2
        while up <= N:
            up = up << 1
            
        return up-1-N